src/hooks/useWindowDimensions.ts

```diff
@@ -0,0 +1,26 @@
+import { useState, useEffect } from "react";
+
+const getWindowDimensions = () => {
+  const { innerWidth: width, innerHeight: height } = window;
+  return {
+    width,
+    height,
+  };
+};
+
+export const useWindowDimensions = () => {
+  const [windowDimensions, setWindowDimensions] = useState(
+    getWindowDimensions()
+  );
+
+  useEffect(() => {
+    function handleResize() {
+      setWindowDimensions(getWindowDimensions());
+    }
+
+    window.addEventListener("resize", handleResize);
+    return () => window.removeEventListener("resize", handleResize);
+  }, []);
```

Comment:
I think this will lead to the creation of multiple `resize` listeners, which makes this hook not reusable

Reply:
yes, but the `return` statement will clean up all of them, and every time we are having the current window size

Reply:
Yes, but imagine using this hook in 5 places simultaneously. It will create 5 listeners

Reply:
make sense, will handle it as well

Reply:
Instead of rolling our own, can use something like https://github.com/streamich/react-use/blob/master/docs/useWindowSize.md

I want to bring this lib in anyway as it has tons of useful util functions like this

Reply:
(Probably not needed here, but for future knowledge, see also: https://developer.mozilla.org/en-US/docs/Web/API/Resize_Observer_API)

Reply:
IMHO `window.innerWidth` deals with it effective.

Reply:
See issues with this method mentioned in https://github.com/sparkletown/sparkle/pull/1311#discussion_r611312150

Reply:
Using `useWindowSize` from `react-use` package